pkg/types/dsse/dsse.go

```diff
@@ -33,5 +42,75 @@ func Validate(ds *pb.DSSERequest) error {
 			return err
 		}
 	}
+	envelope := &dsse.Envelope{}
+	if err := json.Unmarshal([]byte(ds.Envelope), envelope); err != nil {
+		return err
+	}
+	if len(envelope.Signatures) == 0 {
+		return fmt.Errorf("envelope missing signatures")
+	}
+	allPubKeyBytes := make([][]byte, 0)
+	for _, verifier := range ds.Verifier {
+		pubKey := verifier.GetPublicKey()
+		cert := verifier.GetX509Certificate()
+		if pubKey != nil {
+			allPubKeyBytes = append(allPubKeyBytes, pubKey.RawBytes)
+		} else {
+			allPubKeyBytes = append(allPubKeyBytes, cert.RawBytes)
+		}
+	}
+	_, err := verifyEnvelope(allPubKeyBytes, envelope)
+	if err != nil {
+		return err
+	}
 	return nil
 }
+
+// verifyEnvelope takes in an array of possible key bytes and attempts to parse them as x509 public keys.
+// it then uses these to verify the envelope and makes sure that every signature on the envelope is verified.
+// it returns a map of verifiers indexed by the signature the verifier corresponds to.
+// Copied from https://github.com/sigstore/rekor/blob/73dba7c07d0747f00119417fc0ff994a393f97b2/pkg/types/dsse/v0.0.1/entry.go#L364-L403
+func verifyEnvelope(allPubKeyBytes [][]byte, env *dsse.Envelope) (map[string]*x509.PublicKey, error) {
```

Comment:
I really hope this function can be simplified (the underlying EnvelopeVerifier can already handle a set of keys so it should be possible) That said copying existing rekor code seems fine for now.

Reply:
I have an update for this in #144. I think the provider of the dsserequest should provide matching sigs/verifier pairs instead of us having to calculate this on the server side.

Reply:
Will keep this for now and @loosebazooka can remove it when the type is updated